### DISCUSSION TITLE
rules

### DISCUSSION THREAD

**Discussion by davlaw** [arendst/Tasmota#12934] [Q&A] [answered]
I have been having a problem with MJ-S01 Switch as of late (maybe a month or so)

Flashed with default tasmota.bin ( various versions since 9.5

Applied rule per several sites I have seen to make the led change between power on/power off

Rule on power1#state do backlog ledpower1 %value%; ledpower2 %value% endon on power1#boot do backlog ledpower1 %value%; ledpower2 %value% endon

And activated per rule1 1

But console on shows 1 led being acted on, ledpower1, never ledpower2

21:04:32.223 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 24, Data 'ledpower1 1; ledpower2 1'
21:04:32.233 SRC: Backlog
21:04:32.235 CMD: Grp 0, Cmnd 'LEDPOWER', Idx 1, Len 1, Data '1'
21:04:32.240 MQT: mjs01_frontflood/stat/mjs01_frontflood/RESULT = {"LedPower1":"ON"}
21:04:32.484 SRC: Backlog
21:04:32.486 CMD: Grp 0, Cmnd 'LEDPOWER', Idx 2, Len 1, Data '1'
21:04:32.491 MQT: mjs01_frontflood/stat/mjs01_frontflood/RESULT = {"LedPower1":"ON"}
21:04:32.888 CFG: Saved to flash at F7, Count 133, Bytes 4096

I not beyond saying did I miss something, did not read something fully etc.  But I have reset several units, reconfigured them and such but I have missed what ever magic bit that need to be flipped and have been unable to return them to the way they were functioning months ago..

Bug or my ignorance ?

**Comment by sfromis** [comment, marked as answer]
The [command](https://tasmota.github.io/docs/Commands/#ledpower) `LedPower<x>` is described with **Enabled only when LedLink(i) is configured**, and the [default template](https://templates.blakadder.com/martin-jerry-MJ-S01.html) for your device has no LedLink pin.

**Reply by davlaw** [reply]
Well done, I did not read enough it appears (as I eluded to). 

Thanks for your links to find the info on the subject matter
